### PR TITLE
Add valhalla routing

### PR DESCRIPTION
**Pull request by chrstnbwnkl**
I'm creating an PR here for now @nilsnolde so you can have a look first. Basically just reverted the commit that removed mapzen and renamed a few things. One discussion point: I saw that the OSRM attribution links to the about page of the FOSSGIS instance, not the OSRM project itself, so I thought it'd be consistent to do that with Valhalla as well. The article on the FOSSGIS website is in German, however, but I did find that short piece on our website about it, so I'm suggesting to link to that one. Alternatively, we could just link to Valhalla's gh instead.

### CODE REVIEW

File: app/assets/javascripts/osm.js.erb [hunk 1]
```diff
@@ -18,11 +18,15 @@ OSM = {
...
   FOSSGIS_OSRM_URL:        <%= Settings.fossgis_osrm_url.to_json %>,
+  FOSSGIS_VALHALLA_URL:    <%= Settings.fossgis_valhalla_url.to_json %>,
   DEFAULT_LOCALE:          <%= I18n.default_locale.to_json %>,
 
 <% if Settings.key?(:thunderforest_key) %>
   THUNDERFOREST_KEY:       <%= Settings.thunderforest_key.to_json %>,
 <% end %>
+<% if defined?(MAPQUEST_KEY) %>
+  MAPQUEST_KEY:            <%= MAPQUEST_KEY.to_json %>,
+<% end %>
```
Comment (nilsnolde) [inline app/assets/javascripts/osm.js.erb]: this can go, mapquest is kinda dead

File: app/assets/javascripts/index/directions/valhalla.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+function ValhallaEngine(id, costing) {
+  var MZ_INSTR_MAP = [
```
Comment (nilsnolde) [inline app/assets/javascripts/index/directions/valhalla.js]: maybe remove the `MZ_`

**Review by nilsnolde — CHANGES_REQUESTED** [review]
Thanks, looks good, just a few nits. And seeing if we can make the maneuver mapping as similar to OSRM, then we can PR to upstream.

File: app/assets/javascripts/osm.js.erb (hunk 1 above)
Comment (chrstnbwnkl) [reply app/assets/javascripts/osm.js.erb]: whoops, thought I'd removed them all

**Pull request closed without merging**